awx/api/pagination.py

```diff
@@ -18,12 +18,24 @@ def get_next_link(self):
         url = self.request and self.request.get_full_path() or ''
         url = url.encode('utf-8')
         page_number = self.page.next_page_number()
-        return replace_query_param(url, self.page_query_param, page_number)
+        return replace_query_param(self.cap_page_size(url), self.page_query_param, page_number)
 
     def get_previous_link(self):
         if not self.page.has_previous():
             return None
         url = self.request and self.request.get_full_path() or ''
         url = url.encode('utf-8')
         page_number = self.page.previous_page_number()
```

Comment:
Sorry I was meaning to review this, but it took some time.

For this kind of fix, without a doubt I would like to see some tests. For the construction of the test, I would see about either patching the `MAX_PAGE_SIZE` setting or the property of this class. Then you could trigger page size limit with 2 or 3 items, making it a very contained test. That would be an excellent test to have, and whatever solution you go with it'll back it up well.

I think what you have here could work, but I have a lot of questions. I know this is quite complicated, because there are multiple classes coming from Django and DRF that do a lot of the same thing. After poking around those, I would like to inspect things like `self.page.page_size` in contexts like this. This might also be a valid bug in Django rest framework, and having more information would let us say one way or the other. Having the test (probably in `api/main/tests/functional/api`) will help to drop a `import pdb; pdb.set_trace()` and dig into those details.